dotCMS/src/main/java/com/liferay/util/FileUtil.java

```diff
@@ -153,7 +153,11 @@ private static void validateEmptyFile(File source) throws IOException{
         }
 	}
 
-	public static void copyFile(File source, File destination, boolean hardLinks) throws IOException {
+    public static void copyFile(File source, File destination, boolean hardLinks) throws IOException {
+        copyFile(source, destination, hardLinks, true);
+    }
+
+	public static void copyFile(File source, File destination, boolean hardLinks, boolean validateEmptyFile) throws IOException {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 18 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "validateEmptyFile" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2301)